.github/workflows/source.yml

```diff
@@ -0,0 +1,58 @@
+name: source
```

Comment:
We don't need this target except when we make deployment. It just create an archive using `git ls-files` list. I don't know how to break this process by a change in tarantool so unlikely it worth to test it.

So, we don't need to run it on PRs and pushes to short-term branches. We don't need `--init` and artifacts uploading, because we don't run testing here. We don't need  `update_repo.sh` dependencies, because it is not used here.